src/deadline/job_attachments/bucket_sweeper/job_attachments_sweeper.py

```diff
@@ -41,11 +51,50 @@ def __init__(
         self.s3 = s3_client
         self.s3_control = s3_control_client
         self.deadline = deadline_client
+        self.retention_record_handler = retention_record_handler
         self.farm_id = farm_id
         self.account_id = account_id
         self.role_arn = role_arn
         self.bucket_name = bucket_name
 
+    def get_attachments_to_retain(self, queue_job_id_map: Dict[str, List[str]]) -> Set[str]:
+        """
+        Retrieves a set of S3 object keys that should be retained based on queue and job IDs.
+
+        This method queries the retention record handler to get retention records for the specified
+        queue and job IDs, then extracts the unique S3 object keys from those records. The returned
+        set contains S3 object keys that should be retained during cleanup operations.
+
+        Args:
+            queue_job_id_map: A dictionary mapping queue IDs to lists of job IDs. For each queue ID,
+                            the associated job IDs will be used to find retention records.
+                Example: { "queue-1": ["job-1"], "queue-2": ["job-2"] }
+
+        Returns:
+            Set[str]: A set of unique S3 object keys that should be retained.
+
+        Raises:
+            JobAttachmentsSweeperError: If there's an error retrieving retention records from the
+                                    record handler.
+
+        Note:
+            If the queue_job_id_map is empty or if no retention records are found for the specified
+            queue and job IDs, an empty set will be returned.
+        """
+        retain_object_keys: Set[str] = set()
+
+        try:
+            records: List[RetentionRecord] = self.retention_record_handler.get_retention_records(
+                queue_job_id_map=queue_job_id_map
+            )
+        except RetentionRecordHandlerError as err:
+            raise JobAttachmentsSweeperError(message=f"Failed to get retention records: {str(err)}")
+
+        for record in records:
+            retain_object_keys.add(record.s3_object_key)
```

Comment:
nit - Set comprehensions are generally considered more Pythonic and can be slightly more efficient
since they avoid the repeated method calls to add()

```suggestion
       retain_object_keys: Set[str] = {record.s3_object_key for record in records}
```